client/tests/auth.spec.ts

```diff
@@ -0,0 +1,118 @@
+    await page.getByPlaceholder('Name').fill('test');
+    await page.getByPlaceholder('Name').press('Tab');
+    await page.getByPlaceholder('Email').fill('test@gmail.com');
+    await page.getByPlaceholder('Password').click();
+    await page.getByPlaceholder('Password').fill('Gg1234!');
+    await page.getByRole('button', { name: 'Register' }).click();
+    await expect(page.getByText('passwords must be at least 8 characters and contain 1 uppercase, 1 lowercase, 1 number and 1 symbols')).toBeVisible();
+});
+
+test('test user already exists', async ({ page }) => {
+    await page.goto('http://localhost:5173/register');
+    await page.getByPlaceholder('Name').click();
+    await page.getByPlaceholder('Name').fill('test');
+    await page.getByPlaceholder('Name').press('Tab');
+    await page.getByPlaceholder('Email').fill('ofek@gmail.com');
+    await page.getByPlaceholder('Password').click();
+    await page.getByPlaceholder('Password').fill('Gg1234!');
+    await page.getByRole('button', { name: 'Register' }).click();
+    await expect(page.getByText('User already exists')).toBeVisible();
+});
+
+test('test new register success', async ({ page }) => {
+    await page.goto('http://localhost:5173/register');
+    await page.getByPlaceholder('Name').click();
+    await page.getByPlaceholder('Name').fill('playwright test');
+    await page.getByPlaceholder('Name').press('Tab');
+    await page.getByPlaceholder('Email').fill(`${Math.random().toString(36).slice(2)}@gmail.com`);
+    await page.getByPlaceholder('Password').click();
+    await page.getByPlaceholder('Password').fill('Gg12345!');
+    await page.getByRole('button', { name: 'Register' }).click();
+    await expect(page.getByRole('link', { name: 'playwright test' })).toBeVisible();
+});
+
+test('test wrong password', async ({ page }) => {
+    await page.goto('http://localhost:5173/login');
+    await page.getByPlaceholder('Email').fill('test@gmail.com');
+    await page.getByPlaceholder('Password').fill('Ggggggg3');
+    await page.getByRole('button', { name: 'login' }).click();
+    await expect(page.getByText('Invalid email or password')).toBeVisible();
+});
+test('test wrong email', async ({ page }) => {
+    await page.goto('http://localhost:5173/login');
+    await page.getByPlaceholder('Email').fill('te@gmail.com');
+    await page.getByPlaceholder('Password').fill('Gg12345!');
+    await page.getByRole('button', { name: 'login' }).click();
+    await expect(page.getByText('Invalid email or password')).toBeVisible();
+});
+
+test('test login success', async ({ page }) => {
+    await page.goto('http://localhost:5173/login');
+    await page.getByPlaceholder('Email').fill('test@gmail.com');
+    await page.getByPlaceholder('Password').click();
+    await page.getByPlaceholder('Password').fill('Gg12345!');
+    await page.getByRole('button', { name: 'Login' }).click();
+    await expect(page.getByRole('link', { name: 'test' })).toBeVisible();
+
+});
+
```

Comment:
Create a utils file, and move all helpers function into it, see where you can use the same code and make it a general function.
If it's relevant for a specific file it goes to the top of the file, if for more it goes to utils.
Functions like generate password or generate message are relevant as a util.
Create also teardown function running after tests that cleans everything and closes.